contrib/epee/include/net/net_helper.h

```diff
@@ -203,13 +233,14 @@ namespace net_utils
 				{
 					m_connected = false;
 					m_socket.shutdown(boost::asio::ip::tcp::socket::shutdown_both);
-					
+					boost::system::error_code ec;
+					m_ssl_socket.shutdown(ec);
```

Comment:
Does this need to be done first ? It might be OK, but ssl seems to be something higher level than the socket. Ignoring the error code just means you don't really catch those kind of errors. Just printing and otherwise ignoring still leaves a trace.